cou/steps/plan.py

```diff
@@ -596,7 +595,27 @@ async def _get_upgradable_hypervisors_machines(
     if cli_force:
         return nova_compute_machines
 
-    return await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+    empty_hypervisors = await get_empty_hypervisors(nova_compute_units, analysis_result.model)
+
+    machine_to_unit_name = {
+        machine.machine_id: unit.name
+        for unit, machine in zip(nova_compute_units, nova_compute_machines)
+    }
+    empty_hypervisor_ids = {machine.machine_id for machine in empty_hypervisors}
+    all_hypervisor_ids = {machine.machine_id for machine in nova_compute_machines}
+    skipped_hypervisor_ids = all_hypervisor_ids - empty_hypervisor_ids
```

Comment:
I would prefer doing this on the object level instead of just using the ids of the objects